src/pyhf/tensor/pytorch_backend.py

```diff
@@ -322,8 +324,11 @@ def normal_cdf(self, x, mu=0.0, sigma=1.0):
         Returns:
             PyTorch FloatTensor: The CDF
         """
-        normal = torch.distributions.Normal(mu, sigma)
-        return normal.cdf(x)
+        # the following variant is numerically more stable for low p-values
+        # than the default version that uses torch.erf
+        # erfc(x) = 1 - erf(x)
+        mu, sigma = broadcast_all(mu, sigma)
```

Comment:
```suggestion
        mu, sigma = self.broadcast(mu, sigma)
```

Not sure what is more appropriate here.

Reply:
mhh in principle one could do
```python
mu, sigma = self.simple_broadcast(self.astensor(mu), self.astensor(sigma))
```
I used `broadcast_all`, since it is also done in [torch.Normal](https://github.com/pytorch/pytorch/blob/3bbedb34b9b316729a27e793d94488b574e1577a/torch/distributions/normal.py#L44). There is a slight difference - `broadcast_all` produces scalar Tensors for scalar numbers whereas `self.astensor` produces arrays of length 1. Not sure if this has any impact. The [docstring example for torch.Normal](https://github.com/pytorch/pytorch/blob/3bbedb34b9b316729a27e793d94488b574e1577a/torch/distributions/normal.py#L17) also uses arrays of length 1.

Reply:
I see benefits in either direction.

- minimizing the necessary interface with a backend makes it easier in the long run to maintain...
- broadcast_all is likely faster/more efficient than our broadcast implementations...

Reply:
I would do broadcast_all